src/s_conf.c

```diff
@@ -2226,7 +2226,10 @@ merge_confs()
     merge_classes();        /* this should always be done first */
     merge_me();
     merge_connects();
+
+    MyFree(allows);  // Clear then readd Allow blocks in same order as in conf -Holbrook
```

Comment:
Have you tested this? maybe I'm reading this wrong but if you free allows, wouldn't this leave all the allow objects in the nested list dangling?

Also merge_allows() marks the existing items in the allows list as invalid (legal==-1) and starts again from new_allows. Have you tested the order is maintained? I think the issue happens when a user connects, it's assigned a class based on allow but that assignment isn't following the order that's specified in the conf file. One thing we discussed was to assign an index/order number to the entries so we can start at 1.